### PR TITLE
Fix in writing the neo-event wddx + extra unittest

### PR DESCRIPTION
**Pull request by bferdinandus**

### CODE REVIEW

File: models/BaseAdobe.cfc [hunk 1]
```diff
@@ -1043,9 +1043,9 @@ component accessors=true extends='cfconfig-services.models.BaseConfig' {
 			for (var eventGatewayConfiguration in eventGatewayConfigurations) {
 				var currentEventGatewayConfiguration = {
 					"DESCRIPTION" = eventGatewayConfiguration.description,
-					"KILLONTIMEOUT" = eventGatewayConfiguration.killontimeout,
+					"KILLONTIMEOUT" = javacast("boolean", eventGatewayConfiguration.killontimeout),
 					"CLASS" = eventGatewayConfiguration.class,
-					"STARTTIMEOUT" = eventGatewayConfiguration.starttimeout,
+					"STARTTIMEOUT" = javacast("int", eventGatewayConfiguration.starttimeout),
```
Comment (bdw429s) [inline models/BaseAdobe.cfc]: If you look elsewhere in the file you'll see I'm forcing ints by doing

```js
expression+0
```

File: models/BaseConfig.cfc [hunk 2]
```diff
@@ -993,17 +993,17 @@ component accessors="true" {
...
+	* @gatewayId An event gateway ID to identify the specific event gateway instance.
+	* @type The event gateway type, which you select from the available event gateway types, such as SMS or Socket.
 	* @cfcPaths The absolute path to the listener CFC or CFCs that handles incoming messages.
+	* @mode The event gateway start-up status; one of the following: automatic, manual, disabled
 	* @configurationPath A configuration file, if necessary for this event gateway type or instance.
-	* @gateway Id An event gateway ID to identify the specific event gateway instance.
-	* @mode The event gateway start-up status; one of the following: Automatic, Manual, Disabled
-	* @type The event gateway type, which you select from the available event gateway types, such as SMS or Socket.
 	*/
-	function addGatewayInstance(array cfcPaths,
```
Comment (bdw429s) [inline models/BaseConfig.cfc]: Just curious why these were rearranged?

File: models/BaseConfig.cfc [hunk 3]
```diff
@@ -1012,6 +1012,14 @@ component accessors="true" {
 		}
 
 		var thisEventGatewayInstances=getEventGatewayInstances() ?: [];
+		var i=0;
```
Comment (bdw429s) [inline models/BaseConfig.cfc]: I assume saving the same instance twice wasn't overwriting?

File: models/BaseConfig.cfc (hunk 2 above)
Comment (bferdinandus) [reply models/BaseConfig.cfc]: To have the same order as in the coldfusion administrator.

File: models/BaseAdobe.cfc (hunk 1 above)
Comment (bferdinandus) [reply models/BaseAdobe.cfc]: thanks

File: models/BaseAdobe.cfc [hunk 4]
```diff
@@ -1043,9 +1043,9 @@ component accessors=true extends='cfconfig-services.models.BaseConfig' {
 			for (var eventGatewayConfiguration in eventGatewayConfigurations) {
 				var currentEventGatewayConfiguration = {
 					"DESCRIPTION" = eventGatewayConfiguration.description,
-					"KILLONTIMEOUT" = eventGatewayConfiguration.killontimeout,
+					"KILLONTIMEOUT" = javacast("boolean", eventGatewayConfiguration.killontimeout),
```
[the remark this answers is not in the document]
Comment (bferdinandus) [reply models/BaseAdobe.cfc]: thanks

File: models/BaseConfig.cfc (hunk 3 above)
Comment (bferdinandus) [reply models/BaseConfig.cfc]: Ideed. The gateway instances is "just an array", so you could end up with two gateway instances with the same type. Coldfusion does not like that.

**Pull request merged**